api/scripts/wazuh-apid.py

```diff
@@ -169,19 +275,47 @@ def exit_handler(signum, frame):
     pyDaemonModule.delete_pid(API_MAIN_PROCESS, api_pid)
 
 
+def add_debug2_log_level_and_error():
+    """Add a new debug level used by wazuh api and framework."""
+
+    logging.DEBUG2 = 6
+
+    def debug2(self, message, *args, **kws):
+        if self.isEnabledFor(logging.DEBUG2):
+            self._log(logging.DEBUG2, message, args, **kws)
+
+    def error(self, msg, *args, **kws):
+        if self.isEnabledFor(logging.ERROR):
+            if 'exc_info' not in kws:
+                kws['exc_info'] = self.isEnabledFor(logging.DEBUG2)
+            self._log(logging.ERROR, msg, args, **kws)
+
+    logging.addLevelName(logging.DEBUG2, "DEBUG2")
+
+    logging.Logger.debug2 = debug2
+    logging.Logger.error = error
```

Comment:
There seems to be something wrong. When using this api configuration:
```YAML
 logs:
  level: "debug2"
  format: "plain,json"
  max_size:
    enabled: True
    size: "1M"
```

This error is shown:
```
# /var/ossec/bin/wazuh-apid -f
Traceback (most recent call last):
  File "/var/ossec/framework/python/lib/python3.10/logging/config.py", line 619, in configure
    self.configure_logger(name, loggers[name])
  File "/var/ossec/framework/python/lib/python3.10/logging/config.py", line 796, in configure_logger
    self.common_logger_config(logger, config, incremental)
  File "/var/ossec/framework/python/lib/python3.10/logging/config.py", line 781, in common_logger_config
    logger.setLevel(logging._checkLevel(level))
  File "/var/ossec/framework/python/lib/python3.10/logging/__init__.py", line 198, in _checkLevel
    raise ValueError("Unknown level: %r" % level)
ValueError: Unknown level: 'DEBUG2'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/ossec/api/scripts/wazuh-apid.py", line 388, in <module>
    logging.config.dictConfig(uvicorn_params['log_config'])
  File "/var/ossec/framework/python/lib/python3.10/logging/config.py", line 811, in dictConfig
    dictConfigClass(config).configure()
  File "/var/ossec/framework/python/lib/python3.10/logging/config.py", line 621, in configure
    raise ValueError('Unable to configure logger '
ValueError: Unable to configure logger 'wazuh-api'
```

Reply:
Fixed. The debug level must be created before configuring the logs.

https://github.com/wazuh/wazuh/blob/3672b5ef17f00770e66fa9433eca37834749e870/api/scripts/wazuh-apid.py#L388-L391